alica_test_utility/include/alica/test/TestContext.h

```diff
@@ -90,17 +91,92 @@ class TestContext : public alica::AlicaContext
      * @param id ID of the state.
      * @return True if the state identified by id is currently active, false otherwise.
      */
-    bool isStateActive(int64_t id) const;
+    [[deprecated("Use isStateActive(state_name) instead")]] bool isStateActive(int64_t id) const;
 
     /**
      * Similar to isStateActive(id), this returns true if the
      * agent has an active state in the plan with the given ID.
      * @param id ID of the plan.
      * @return True if plan is active, false otherwise.
      */
-    bool isPlanActive(int64_t id) const;
+    [[deprecated("Use getActivePlan(plan_name) instead")]] bool isPlanActive(int64_t id) const;
+
+    /**
+     * Get the behaviour's runnable object given its name
+     * @param name Either the fully qualified name or just the name of the behaviour
+     * @return Pointer to the runnable object or nullptr if the behaviour is not found or not active
+     * nullptr is also returned if there are multiple behaviours active with the same name. Use the
+     * fully qualified name to retrieve the behaviour in this case
+     * The reason for failure can be retrieved by calling getLastFailure()
+     */
+    BasicBehaviour* getActiveBehaviour(const std::string& name);
+
+    /**
+     * Get the plan's runnable object given its name
+     * @param name Either the fully qualified name or just the name of the plan
+     * @return Pointer to the runnable object or nullptr if the plan is not found or not active
+     * nullptr is also returned if there are multiple plans active with the same name. Use the
+     * fully qualified name to retrieve the plan in this case
+     * The reason for failure can be retrieved by calling getLastFailure()
+     */
+    BasicPlan* getActivePlan(const std::string& name);
+
+    /**
+     * Set a transition defined by the states it connects in the given running plan
+     * This method requires the blackboard key with the name `<in_state_name>2<out_state_name>` to be defined in the plan's
+     * blackboard & the TriggerFromInputCond condition to be used for the transition
+     * @param runningPlanName The name (fully qualified if required) of the plan in which the transition is present. This plan should be active
+     * @param inState The name of the incoming state of the transition
+     * @param outState The name of the outgoing state of the transition
+     */
+    bool setTransitionCond(const std::string& runningPlanName, const std::string& inState, const std::string& outState);
+
+    /**
+     * Reset's a transition. Refer to setTransitionCond for details
+     */
+    bool resetTransitionCond(const std::string& runningPlanName, const std::string& inState, const std::string& outState);
+
+    /**
+     * Reset all transitions in the given plan, notes:
+     * The plan should be active
+     * Typically this method has to be called in all plans that use the TriggerFromInputCond condition in the init
```

Comment:
Shouldn't such plans default to evaluating transitions to false when they init, and so we usually dont need to call this?

Reply:
I guess it should. unordered_map's emplace should value initialize default types to 0 & it does seem to be the case from my tests. So I will remove the calls to this method in the plan's init, but retain the method as a util function.

Reply:
I tried & it does not work at the moment because there is no support for types yet. So the type is std::any & the default constructor initialises it to an empty value. We can remove this when the types PR is merged. I'll add it back in for now